lib/endpoint/loadbalancer/loadbalancer.go

```diff
@@ -0,0 +1,110 @@
+package loadbalancer
+
+import (
+	"context"
+	"github.com/backube/volsync/lib/endpoint"
+	"github.com/backube/volsync/lib/meta"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Endpoint struct {
+	hostname       string
+	ingressPort    int32
+	backendPort    int32
+	namespacedName types.NamespacedName
+	objMeta        meta.ObjectMetaMutation
+}
+
+func (e *Endpoint) NamespacedName() types.NamespacedName {
+	return e.namespacedName
+}
+
+func (e *Endpoint) Hostname() string {
+	return e.hostname
+}
+
+func (e *Endpoint) BackendPort() int32 {
+	return e.backendPort
+}
+
+func (e *Endpoint) IngressPort() int32 {
+	return e.ingressPort
+}
+
+func (e *Endpoint) IsHealthy(c client.Client) (bool, error) {
+	svc := &corev1.Service{}
+	err := c.Get(context.Background(), e.NamespacedName(), svc)
+	if err != nil {
+		return false, err
+	}
+
+	if len(svc.Status.LoadBalancer.Ingress) > 0 {
+		if svc.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			e.hostname = svc.Status.LoadBalancer.Ingress[0].Hostname
+		}
+		if svc.Status.LoadBalancer.Ingress[0].IP != "" {
+			e.hostname = svc.Status.LoadBalancer.Ingress[0].IP
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func NewEndpoint(c client.Client,
+	name types.NamespacedName,
+	metaMutation meta.ObjectMetaMutation,
+	backendPort, ingressPort int32) (endpoint.Endpoint, error) {
+	s := &Endpoint{
+		namespacedName: name,
+		objMeta:        metaMutation,
+		backendPort:    backendPort,
+		ingressPort:    ingressPort,
+	}
+
+	err := s.createService(c)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, err
+}
+
+func (e *Endpoint) createService(c client.Client) error {
```

Comment:
need to change the implementation to CreateOrUpdate for a more consistent reconcile experience